tobac/tracking.py

```diff
@@ -488,126 +419,1020 @@ def linking_trackpy(
+        trajectories_unfiltered["cell"].isin(stub_cell_nums), "cell"
+    ] = cell_number_unassigned
+
+    trajectories_filtered = trajectories_unfiltered
+
+    # Interpolate to fill the gaps in the trajectories (left from allowing memory in the linking)
+    trajectories_filtered_unfilled = deepcopy(trajectories_filtered)
+
+    trajectories_filtered_filled = trajectories_filtered_unfilled
+    trajectories_final = add_cell_time(
+        trajectories_filtered_filled, cell_number_unassigned=cell_number_unassigned
+    )
+    # Add metadata
+    trajectories_final.attrs["cell_number_unassigned"] = cell_number_unassigned
+
+    # add coordinate to raw features identified:
+    logging.debug("start adding coordinates to detected features")
+    logging.debug("feature linking completed")
+    # trajectories_final.reset_index(inplace=True, drop=True)
+
+    return trajectories_final
+
+
+def _clean_track_dfs_for_append(
+    tracks_orig: pd.DataFrame,
+    new_features: pd.DataFrame,
+    memory: int,
+    span: int,
+    cell_number_unassigned: int,
+) -> tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+    """Split up the input tracks and new features into three dataframes:
+    tracks_orig_cut, the input tracks up to (exclusive) of the new dataframe to be tracked,
+    tracks_orig_retrack, the original input tracks needed to retrack with tracks_new
+    tracks_vel_calc, the input tracks needed to calculate cell velocity, and
+    tracks_new, the input tracks that will be tracked by the append function.
+
+    Parameters
+    ----------
+    tracks_orig: pd.DataFrame
+        Original input tracks
+    new_features: pd.DataFrame
+        New features to track
+    memory: int
+        Memory parameter for trackpy
+    span: int
+        Span parameter for trackpy
+    cell_number_unassigned: int
+        Unassigned cell number
+
+    Returns
+    -------
+    tracks_orig_cut, tracks_vel_calc, tracks_new (all pd.DataFrame)
+        tracks_orig_cut, the input tracks up to (exclusive) of the new dataframe to be tracked,
+        tracks_orig_retrack, the original input tracks needed to retrack with tracks_new
+        tracks_vel_calc, the input tracks needed to calculate cell velocity, and
+        tracks_new, the input tracks that will be tracked by the append function.
+
+
+    """
+
+    # need to cut down the existing track array to just the parts we are interested in
+    # for preserving
+    min_frame_orig_needed = max(max(tracks_orig["frame"]) - memory - 1, 0)
+    max_frame_orig_needed = max(tracks_orig["frame"])
+    frames_orig_cut = np.arange(min_frame_orig_needed, max_frame_orig_needed + 1, 1)
+
+    tracks_orig_cut = copy.deepcopy(
+        tracks_orig[~tracks_orig["frame"].isin(frames_orig_cut)]
+    )
+    tracks_orig_retrack = copy.deepcopy(
+        tracks_orig[tracks_orig["frame"].isin(frames_orig_cut)]
+    )
+    tracks_orig_retrack["cell"].replace(cell_number_unassigned, np.nan, inplace=True)
```

Comment:
This line produces the following warning, which will become an exception in future versions of pandas:
```
FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.
```

It also converts the type of the cell column to floating, which I don't think is desirable behaviour. Is there a way of avoiding this?